### PR TITLE
Fixes 2769: Remove select deprecated component

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

- This replaces all deprecated select components with the new patternfly 5 variant.
- This is rebased ontop of [this](https://github.com/content-services/content-sources-frontend/pull/334) PR, and should thus be merged afterward.

Other changes: 

- Small fix to prevent cacheing data when editing a repository and then re-opening the same edit modal for that repository.

## Testing steps

Affects the following: 
- Add/edit modal arch/version dropdowns
- AdminTask Filters
- ContentList Filters
- SnapshotDate selector
- TemplateList filter

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2769

File: src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx [hunk 1]
```diff
@@ -139,30 +139,60 @@ const AdminTaskFilters = ({ isLoading, setFilterData, filterData }: Props) => {
...
+                'data-ouia-component-id': `filter_${status}`,
+              })) as SelectOptionProps[]
+            }
             isDisabled={isLoading}
-            options={statusValues}
-            variant={SelectVariant.checkbox}
-            selectedProp={selectedStatuses}
-            setSelected={setSelectedStatuses}
-            placeholderText='Filter by status'
+            menuValue='Filter by Status'
```
Comment (xbhouse) [inline src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx:165]: sentence case 🙈

File: src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx [hunk 2]
```diff
@@ -139,30 +139,60 @@ const AdminTaskFilters = ({ isLoading, setFilterData, filterData }: Props) => {
...
         return (
-          <DropdownSelect_Deprecated
-            toggleAriaLabel='filter type'
-            toggleId='typeSelect'
-            ouiaId='filter_type'
+          <DropdownSelect
+            onSelect={(_, val) =>
+              setSelectedTypenames((prev) =>
+                selectedTypenames.includes(val as string)
+                  ? [...prev.filter((item) => item !== (val as string))]
```
Comment (xbhouse) [inline src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx:175]: do we need a new array here? it seems to function the same as the type filter which doesn't use an extra array

File: src/Pages/Repositories/ContentListTable/components/AddContent/helpers.test.ts [hunk 3]
```diff
@@ -24,100 +24,91 @@ it('isValidURL', () => {
...
-        skipped: false,
-        valid: true,
-        error:
-          'Error fetching YUM metadata: Head "https://bobjull.co": dial tcp: lookup bobjull.co: no such host',
-        http_code: 0,
-        metadata_present: false,
-        metadata_signature_present: false,
-      },
+it('mapValidationData, ensure url error shows', () => {
+  //   expect(mapValidationData({}, {})).toEqual({});
```
Comment (xbhouse) [inline src/Pages/Repositories/ContentListTable/components/AddContent/helpers.test.ts:55]: can this commented out code be removed? looks like you're using it in a test it a bit further down

File: src/Pages/Repositories/ContentListTable/components/ContentListFilters.tsx [hunk 4]
```diff
@@ -189,44 +188,83 @@ const ContentListFilters = ({
...
+              children: arch,
+              'data-ouia-component-id': `filter_${arch}`,
+            }))}
             isDisabled={isLoading}
-            options={Object.keys(archNamesLabels)}
-            variant={SelectVariant.checkbox}
-            selectedProp={selectedArches}
-            setSelected={setSelectedArches}
-            placeholderText='Filter by architecture'
+            menuValue='Filter by Arch'
```
Comment (xbhouse) [inline src/Pages/Repositories/ContentListTable/components/ContentListFilters.tsx:239]:
```suggestion
            menuValue='Filter by architecture'
```

File: src/Pages/Repositories/ContentListTable/components/ContentListFilters.tsx [hunk 5]
```diff
@@ -189,44 +188,83 @@ const ContentListFilters = ({
...
+              children: status,
+              'data-ouia-component-id': `filter_${status}`,
+            }))}
             isDisabled={isLoading}
-            options={statusValues}
-            variant={SelectVariant.checkbox}
-            selectedProp={selectedStatuses}
-            setSelected={setSelectedStatuses}
-            placeholderText='Filter by status'
+            menuValue='Filter by Status'
```
Comment (xbhouse) [inline src/Pages/Repositories/ContentListTable/components/ContentListFilters.tsx:266]:
```suggestion
            menuValue='Filter by status'
```

File: src/Pages/Repositories/ContentListTable/components/SnapshotDetailsModal/SnapshotSelector.tsx [hunk 6]
```diff
@@ -50,15 +52,23 @@ export function SnapshotSelector() {
         aria-label='snapshot-selector'
         fieldId='snapshot'
       >
-        <DropdownSelect_Deprecated
+        <DropdownSelect
+          onSelect={(_, val) => setSelected(val as string)}
+          selected={uuidMapper[snapshotUUID]}
+          menuToggleProps={{
+            className: classes.minWidth,
+            'aria-label': 'filter architecture',
```
Comment (xbhouse) [inline src/Pages/Repositories/ContentListTable/components/SnapshotDetailsModal/SnapshotSelector.tsx:60]:
```suggestion
            'aria-label': 'filter date',
```

File: src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx [hunk 7]
```diff
@@ -0,0 +1,339 @@
+import { useEffect, useMemo, useState } from 'react';
```
Comment (xbhouse) [inline src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx:1]: must have missed this in the upload PR, typo in this directory name

File: src/Pages/Templates/TemplatesTable/components/TemplateFilters.tsx [hunk 8]
```diff
@@ -66,7 +65,7 @@ const Filters = ({ isLoading, setFilterData, filterData }: Props) => {
     queryClient.getQueryData<RepositoryParamsResponse>(REPOSITORY_PARAMS_KEY) || {};
 
   const clearFilters = () => {
-    setFilterType('Name');
+    // setFilterType('Name');
```
Comment (xbhouse) [inline src/Pages/Templates/TemplatesTable/components/TemplateFilters.tsx:68]: there are a few commented out lines in this file, can they be removed?

**Comment by xbhouse** [comment]
i think `any` should be capitalized in the Versions column here, and i don't think we should be able to select multiple versions on the add/edit repo modals. maybe fixed with a rebase?

Screenshot 2024-09-24 at 4 56 42 PM
Screenshot 2024-09-24 at 4 58 27 PM

File: src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx (hunk 1 above)
Comment (Andrewgdewar) [reply src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx:165]: That was dangerous.. Maria would could have seen this... 😅

File: src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx (hunk 7 above)
Comment (Andrewgdewar) [reply src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx:1]: Compontents....

File: src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx (hunk 2 above)
Comment (Andrewgdewar) [reply src/Pages/Repositories/AdminTaskTable/components/AdminTaskFilters.tsx:175]: Hmm not sure why I did that, good call!

**Comment by xbhouse** [comment]
> i think `any` should be capitalized in the Versions column here, and i don't think we should be able to select multiple versions on the add/edit repo modals. maybe fixed with a rebase?
> 
> Screenshot 2024-09-24 at 4 56 42 PM Screenshot 2024-09-24 at 4 58 27 PM
> 
> Rebased, issue identified with 'any' addressed there.
> 
> AFAIK, multiple versions have always been selectable, I think this was due to the two fields being filter references?
> 
> I had similar questions and @jlsherrill explained the reasoning to me once.

ohp you're right.. i got confused because i looked in main and didn't see the checkboxes so assumed we couldn't multiselect, but we can :) disregard that comment 😅

**Review by xbhouse — APPROVED** [review]
tested and all looks good!

**Comment by swadeley** [comment]
Hi @Andrewgdewar , please rebase now that #334 is merged.

**Comment by swadeley** [comment]
Thank you

**Pull request merged**